bpo-28180: assume UTF-8 for Mac OS X PEP 538 tests

ncoghlan opened the pull request.

ncoghlan committed:
PEP 538: assume UTF-8 for Mac OS X tests

ncoghlan commented:
@haypo, @ned-deily, @ronaldoussoren I think this should fix most of the current PEP 538 test failures on Mac OS X, except for the "LANG=UTF-8" one, where I'm not sure of the actual expected behaviour (I would have expected that to work, as even though `UTF-8` is only defined as a partial locale, we're also only trying to configure it for `LC_CTYPE`).

(I haven't submitted it to the custom buildbot branch, as I'm about to head to bed, and don't want to claim the branch overnight).

vstinner commented:
test_locale_c_coercion still fail on macOS with this change. Only FS encoding is always utf-8 on macOS, the locale encoding is still impacted by LC_ALL, LC_CTYPE and LANG.

ncoghlan committed:
bpo-28180: Further PEP 538 test case fixes

- clarify it's the default FS encoding we're checking
- don't skip the whole module when no target locales are available
- skip the explicit LANG=UTF-8 behaviour for now

ncoghlan committed:
Use correct name for test fixture method

ncoghlan commented:
It looks like it's going to take the custom Mac OS X builder a while to catch up, so I'm going to go ahead and merge this, since it's at least less wrong than what's currently checked in (even though it doesn't fix the problems on FreeBSD).
